File: programmers/level1/제일_작은_수_제거하기.java
package programmers;

import java.util.Arrays;

public class 제일_작은_수_제거하기 {
	static
	class Solution {
		  public int[] solution(int[] arr) {
			  if(arr.length==1) {
				  return new int[] {-1};
			  }
			  int minNum = Integer.MAX_VALUE;
		      int minIdx = -1;
		      for(int i = 0; i < arr.length; ++i) {
		    	  if(minNum > arr[i]) {
		    		  minNum = arr[i];
		    		  minIdx = i;
		    	  }
		      }
		      int[] answer = new int[arr.length-1];
		      for(int i = 0, arrIdx = 0; arrIdx < arr.length; ++arrIdx) {
		    	  if(arrIdx != minIdx) {
		    		  answer[i++] = arr[arrIdx];
		    	  }
		      }
		      return answer;
		  }
		}
	public static void main(String[] args) {
		int[] arr = {4,3,2,1};
		Solution sol = new Solution();
		System.out.println(Arrays.toString(sol.solution(arr)));
	}
}

File: programmers/js/new/level1/제일_작은_수_제거하기.js
function solution(arr) {
    let min = Number.MAX_VALUE;
    let minIdx = -1;
    arr.forEach((num, idx) => {
        if (num < min) {
            min = num;
            minIdx = idx;
        }
    });
    arr.splice(minIdx, 1);

    if (arr.length === 0) {
        arr.push(-1);
    }
    return arr;
}
